File: pset6/readability/readability.py
string = input("Text: ")
lettercount = 0
wordcount = 1 #starting count from 1 because last word is not accounted for
sentencecount = 0

#iterating across entire string and checking char type
for char in string:
    if char.isalpha():
        lettercount += 1
    elif char == " ":
        wordcount += 1
    elif char == "." or char == '!' or char == '?':
        sentencecount += 1

#calculating index
L = (float(lettercount) / float(wordcount)) * 100.0
S = (float(sentencecount) / float(wordcount)) * 100.0
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Below Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int indexCalc(float letterCount, float wordCount, float sentCount)
{
    float L = (letterCount / wordCount) * 100;
    float S = (sentCount / wordCount) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int roundedIndex = round(index);
    return roundedIndex;
}

int main(int argc, string argv[])
{
    int lettercount = 0;
    //initializing word counter as 1 because the for loop does not account for the last word
    int wordcount = 1;
    int sentencecount = 0;
    string input = get_string("Text: ");
    //iterates over string and passes each char through a series of checks
    for (int i = 0; i < strlen(input); i++)
    {
        //letter check
        //if passes, skips the rest of the checks
        if (isalpha(input[i]))
        {
            lettercount++;
        }
        else //if char is not letter
        {
            switch (input[i])
            {
                case ' ': //check for spaces, spaces = words
                    wordcount++;
                    break;
                case '.': //next three cases check for sentence ending chars
                    sentencecount++;
                    break;
                case '!':
                    sentencecount++;
                    break;
                case '?':
                    sentencecount++;
                    break;
            }
        }
    }

    int index = indexCalc((float)lettercount, (float)wordcount, (float)sentencecount);
    printf("Letters: %i\nWords: %i\nSentences: %i\n", lettercount, wordcount, sentencecount);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 15)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}